inc/Engine/Common/Database/Queries/AbstractQuery.php

```diff
@@ -45,7 +45,7 @@ public function get_row( string $url, bool $is_mobile = false ) {
 	 *
 	 * @param int $row_id DB Row ID.
 	 *
-	 * @return array|false
+	 * @return array|bool|object false if no row found, array if row found.
```

Comment:
I did put object because of this function: 
https://github.com/wp-media/wp-rocket/blob/7cab3aa6e7a3d8566e4f8abb7e575f7f9d4c4a50/inc/Engine/Optimization/RUCSS/Cron/Subscriber.php#L53-L61

To be honest it did confused me a bit. We are checking if there is an object in the result of that function, I don't get why as it should be an array. And I was feeling strange we were bailing out for all these times. 
Do you have an explanation ?

Reply:
the method `get_Row_by_id()` returns either false, or the value of the last element of the array returned by the `query()` method, since it uses `array_pop()` to do that.

So it's likely the accurate typehint is `object|false`